tools/fp20compiler/rc1.0_final.cpp

```diff
@@ -76,39 +88,20 @@ void FinalCombinerStruct::Validate()
         rgb.d = temp;
     }
 
-    if (REG_SPARE0_PLUS_SECONDARY_COLOR == rgb.a.reg.bits.name ||
-        REG_DISCARD == rgb.a.reg.bits.name ||
-        REG_DISCARD == rgb.b.reg.bits.name ||
-        REG_DISCARD == rgb.c.reg.bits.name ||
-        REG_DISCARD == rgb.d.reg.bits.name)
-        errors.set("invalid input register for final rgb");
-
-    if (RCP_BLUE == rgb.a.reg.bits.channel ||
-        RCP_BLUE == rgb.b.reg.bits.channel ||
-        RCP_BLUE == rgb.c.reg.bits.channel ||
-        RCP_BLUE == rgb.d.reg.bits.channel)
-        errors.set("blue register used in final rgb");
-
-    if ((REG_E_TIMES_F == rgb.a.reg.bits.name ||
-        REG_E_TIMES_F == rgb.b.reg.bits.name ||
-        REG_E_TIMES_F == rgb.c.reg.bits.name ||
-        REG_E_TIMES_F == rgb.d.reg.bits.name) && !hasProduct)
-        errors.set("final_product used but not set");
-
-    if (RCP_NONE == rgb.a.reg.bits.channel)
-        rgb.a.reg.bits.channel = RCP_RGB;
-    if (RCP_NONE == rgb.b.reg.bits.channel)
-        rgb.b.reg.bits.channel = RCP_RGB;
-    if (RCP_NONE == rgb.c.reg.bits.channel)
-        rgb.c.reg.bits.channel = RCP_RGB;
-    if (RCP_NONE == rgb.d.reg.bits.channel)
-        rgb.d.reg.bits.channel = RCP_RGB;
-    if (RCP_NONE == product.e.reg.bits.channel)
-        product.e.reg.bits.channel = RCP_RGB;
-    if (RCP_NONE == product.f.reg.bits.channel)
-        product.f.reg.bits.channel = RCP_RGB;

```

Comment:
E and F channel initialization was previously unconditional, it is now conditional (as the call to `ValidateRgbInputRegister` which does it is conditional).
So, depending on the emitter it might cause issues later.

The condition should be moved into the function (or removed? Is it even necessary?).